staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -48,3 +48,47 @@ type AdmissionPluginConfiguration struct {
 	// +optional
 	Configuration *runtime.Unknown `json:"configuration"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConnectivityServiceConfiguration provides versioned configuration for network proxy clients.
+type ConnectivityServiceConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// ConnectionServices contains a list of network proxy client configurations
+	ConnectionServices []ConnectionService `json:"connectionServices"`
+}
+
+// ConnectionService provides the configuration for a single network proxy client.
+type ConnectionService struct {
+	// Name is the name of the connectivity service.
+	// Currently supported values are "Master" and "Cluster"
+	Name string `json:"name"`
+
+	// Connection is the exact information used to configure the network proxy
+	Connection Connection `json:"connection"`
+}
+
+// ConnectivityServiceConfiguration provides the configuration for a single network proxy client.
+type Connection struct {
+	// Type is the type of connection used to connect from client to network/proxy-server.
+	// Currently supported values are "http-connect" and "direct".
+	Type string `json:"type"`
+
+	// URL is the location of the proxy server to connect to.
+	// As an example it might be "https://127.0.0.1:8131"
+	// +optional
```

Comment:
if it's optional, you need to describe what happens if it isn't specified in the doc.

Reply:
Agreed. The optional is primarily for the "direct" use case which will not have things like URL and cert materials.
Will specify an error from validation if incorrectly configured.